src/pages/Card/_components/CategoryBox.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import { CARD_DATA } from '@/pages/Card/_constants/cardData';
+import { CATEGORY_BOX_DATA } from '@/pages/Card/_constants/cardData';
+
+import CardBox from './CardBox';
+
+interface CategoryBoxProps {
+  categoryBoxTitle: string;
+}
+
+function CategoryBox({ categoryBoxTitle }: CategoryBoxProps) {
+  const [groupedCards, setGroupedCards] = useState<{
+    [key: string]: (typeof CARD_DATA)[0]['card'];
+  }>({});
+
+  useEffect(() => {
+    const categoryData = CARD_DATA.find((category) => category.category === categoryBoxTitle);
+    if (categoryData) {
+      const grouped = categoryData.card.reduce(
+        (acc, card) => {
+          const tag = card.tags[1]; // 태그의 두 번째 요소 사용
+          if (!acc[tag]) {
+            acc[tag] = [];
+          }
+          acc[tag].push(card);
+          return acc;
+        },
+        {} as { [key: string]: (typeof CARD_DATA)[0]['card'] },
+      );
+
+      setGroupedCards(grouped);
+    }
+  }, [categoryBoxTitle]);
```

Comment:
1. 안에 로직을 따로 함수로 분리한다면 다른 사람이 볼 때 함수명을 보고 어떤 함수인지 알기도 쉽고, 좀 더 재사용성이 높은 코드가 될것 같아유
2. 그리고 불변성을 지키기위해 아래와 같이 스프레드 연산자를 사용하면 어떨까 싶어요! 그리고 삼항 연산자 사용은 한번 제안해봤어요 ~~
```suggestion
const categoryData = CARD_DATA.find((category) => category.category === categoryBoxTitle);
    if (categoryData) {
      const grouped = categoryData.card.reduce(
        (acc, card) => {
          const tag = card.tags[1]; // 태그의 두 번째 요소 사용
          acc[tag] = acc[tag] ? [...acc[tag], card] : [] // 삼항 연산자와 스프레드 연산자 사용
          
          // or 만약 card.tags[1]이 항상 존재하며, card.tags[1]만을 배열 요소 사용한다면 if문 사용 없이 다음과 같이 사용하 는것도 제안
          // acc[tag] = [card] 

          return acc;
        },
        {} as { [key: string]: (typeof CARD_DATA)[0]['card'] },
      );

      setGroupedCards(grouped);
```

Reply:
함수로 따로 분리하는게 가독성이 떠 좋을 거 같네요! 좋은 의견 감사합니다! 삼항 연산자와 스프레드 연산자 적용도 적용하겠습니다!